src/Framework/Features.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Build.Framework
+{
+    /// <summary>
+    /// The status of a feature.
+    /// </summary>
+    public enum FeatureStatus
+    {
+        /// <summary>
+        /// The feature is not found.
+        /// </summary>
+        NotFound,
+
+        /// <summary>
+        /// The feature is available.
+        /// </summary>
+        Available,
+
+        /// <summary>
+        /// The feature is not available.
+        /// </summary>
+        NotAvailable,
+
+        /// <summary>
+        /// The feature is in preview.
+        /// </summary>
+        Preview,
```

Comment:
How do we expect this to be used? For instance suppose we have analyzer API v0.1, and someone writes an analyzer against it. Then we make a breaking API change to 0.2. If the API consumer wishes to support both versions, can they tell which to use? If they support only v0.1, how do they detect that the current MSBuild provides only v0.2?

Reply:
I guess, the name of the feature will reflect the version of the preview changes. `Preview` status just shows that this feature is not stable and will likely change. @rokonec , @baronfel correct me please.

Reply:
>Then we make a breaking API change to 0.2. If the API consumer wishes to support both versions, can they tell which to use?
I suppose they would have to use two different msbuild versions.

Reply:
I planned it mostly with idea that feature is in preview so VS or others can respect their user preference about using or not using preview features. If API changes I was thinking about hierarchical features names like `ANALYZERS.DISTRIBUTED.V1.2` where user can query any level like `ANALYZERS.DISTRIBUTED` for backward compatibility.

Reply:
@rokonec, do you I understand you correctly that instead of making breaking changes to analyzers we will add new API for backward compatibility while in preview phase?

Reply:
> we will add new API for backward compatibility while in preview phase
It is valid option if needed. We can make breaking changes during preview and feature API can signalize consumers state of that new version. If customer needs to know if give MSBuild us preview version 1 or 2 Feature API will tell.

Reply:
I'd like to see an example of this. We have an upcoming "progress" API for tasks which seems like a good use case. Can you make up a feature  name for that and then show an example of what code a task that wishes to provide progress should use to wrap the progress-logging calls?

Reply:
This is all hypothetical as progress API might not be best practical candidate for it. But I can imagine this use case:
Lets have two actors Alice and Bob (they just moved from security team to MSBuild team). Alice tell Bob that she is about to create Progress API with feature name "Logging.Progress.API" and it will be in preview mode soon. Bob would like to use this feature but no sooner than it is stable.
1) Alice is working on Logging API, creates preview version of it and makes feature `Logging.Progress.API => preview`
2) Alice tell Bob that Logging API preview feature is merged to main and point out to code and/or its docs
3) Bob creates branch (or rebase) from main and update related tasks to use progress API, but with runtime conditions `Logging.Progress.API == available`. He will test it locally by hardcoding `Logging.Progress.API into available` and eventually merge it into main
4) Meanwhile Alice will continue with Progress API and eventually consider it stable and change feature status to "available"
5) This will make task modified at 3) finally using Progress API